Automattic/wp-calypso
# Checkout: Update colors, spacing, and typography for new design

## Proposed Changes

This PR modifies the appearance of checkout in relatively small ways (although the visual effect is considerable) to match an updated design. This should not really change any functionality; checkout's step behavior should remain the same before and after this change.

This is part of a larger project of redesigning checkout: p9Jlb4-6FB-p2 Specifically this is the first milestone: pbOQVh-3pj-p2

## Screenshots

### Before

<img width="1280" alt="Screenshot 2023-06-20 at 6 49 42 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/6de9716f-b583-41c1-8761-c199f1f922bd">

<img width="1275" alt="jetpack-before" src="https://github.com/Automattic/wp-calypso/assets/2036909/605b9e52-2786-4378-99fc-a11ca9916d34">

### After

<img width="1279" alt="Screenshot 2023-06-20 at 6 26 35 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/1a4c0435-6a39-4c4a-8219-daab01e5b4fd">

<img width="1273" alt="jetpack-after" src="https://github.com/Automattic/wp-calypso/assets/2036909/b47aef9e-64be-4f0a-a45c-284e49f061b7">


## Testing Instructions

This is a hard thing to test completely because checkout has so many variations. This should not change any functionality, so the first thing is to make sure that checkout renders successfully. Beyond that, it's a good idea to test any variation you can think of. Here are some suggestions:

- Checkout (and any of the other variations) at wide ("desktop") widths.
- Checkout (and any of the other variations) at around 900px wide ("tablet") widths.
- Checkout (and any of the other variations) at around 500px wide ("mobile") widths.
- Checkout for a Jetpack product.
- Checkout for an Akismet product.
- Checkout for a Gift product.
- Checkout with and without having a coupon added.
- Checkout with and without a domain in the cart.
- Checkout with an empty cart.
- Checkout after completing a synchronous (ie: non-3DS credit card) payment before being redirected.
- Checkout with a third-party-developer (eg: Yoast) plugin in the cart.
- Checkout in an RTL locale.
- Checkout with a monthly, yearly, and biennial WPCom plan in the cart.

Some non-checkout things that should be tested as well because they may be affected by this:

- Adding a payment method to an existing subscription.
- Viewing the mini-cart in the masterbar.
- Viewing the one-click checkout upsell after checkout completes.
